src/lib/palette.typ

```diff
@@ -1,36 +1,120 @@
-/// Define a new palette
+#let base-style = (stroke: (paint: black), fill: none)
+
+/// Create a new palette based on a base style
 ///
 /// A palette is a function in the form `index -> style` that takes an
 /// index (int) and returns a canvas style dictionary. If passed the
 /// string `"len"` it must return the length of its styles.
 ///
-/// - stroke (stroke): Single stroke style.
-/// - fills (array): List of fill styles.
-/// -> function
-#let new(stroke, fills) = {
-  if type(fills) != array {fills = (fills,)}
+/// - base (style): Style dictionary to use as base style
+/// - colors (none, array): List of colors to use as stroke and fill color or none
+/// - patterns (none, array): List of stroke patterns to use or none
+/// - stroke (bool): Use color array for stroke paint
+/// - stroke-lighten (ratio): Lighten to apply to the stroke color
+/// - fill (bool): Use color array for fill paint
+/// - fill-lighten (ratio): Lighten to apply to the fill color
+/// -> function Palette function that returns a style for an index
+#let new(base: base-style, colors: (), patterns: (),
+         fill: true, fill-lighten: 0%,
+         stroke: false, stroke-lighten: 0%) = {
+  if not "stroke" in base { base.stroke = (paint: black, thickness: 1pt, dash: "solid") }
+  if not "fill" in base { base.fill = none }
+
+  let color-n = colors.len()
+  let pattern-n = patterns.len()
   (index) => {
```

Comment:
Should palette functions have named arguments for some of their options (`fill:` and `stroke:`) so they can be used via `.with(...)`? @JamesxX

Reply:
I think so, as long as it doesn't add too much overhead, it would be a really nice way to quickly touch up a palette